opentelemetry-sdk/benches/metric_counter.rs

```diff
@@ -0,0 +1,73 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use opentelemetry::{
+    metrics::{Counter, MeterProvider as _},
+    KeyValue,
+};
+use opentelemetry_sdk::metrics::{ManualReader, SdkMeterProvider};
+use rand::{rngs::SmallRng, Rng, SeedableRng};
+
+// Run this benchmark with:
+// cargo bench --bench metric_counter --features=metrics,testing
+fn create_counter() -> Counter<u64> {
+    let meter_provider: SdkMeterProvider = SdkMeterProvider::builder()
+        .with_reader(ManualReader::builder().build())
+        .build();
+    let meter = meter_provider.meter("benchmarks");
+    let counter = meter.u64_counter("counter_bench").init();
+    counter
+}
+
+fn criterion_benchmark(c: &mut Criterion) {
+    counter_no_attributes(c);
+}
+
+fn counter_no_attributes(c: &mut Criterion) {
```

Comment:
nit - this method is recording measurement with attributes, even though the method name has `no_attributes`.